xinference/model/embedding/core.py

```diff
@@ -595,6 +595,60 @@ def encode(
 
             return all_embeddings, all_token_nums
 
+        @no_type_check
+        def _encode_clip(
+            model: SentenceTransformer,
+            sentences: Union[str, List[str]],
+            convert_to_numpy: bool = True,
+            **kwargs,
+        ):
+            import base64
+
+            # import re
+            from io import BytesIO
+
+            from PIL import Image
+
+            def base64_to_image(base64_str: str) -> Image.Image:
+                # base64_data = re.sub("^data:image/.+;base64,", "", base64_str)
+                base64_data = base64_str.split(",", 1)[1]
+                byte_data = base64.b64decode(base64_data)
+                image_data = BytesIO(byte_data)
+                img = Image.open(image_data)
+                return img
+
+            texts = sentences[:-1]
```

Comment:
Why can only the last of sentences be image?

Reply:
Good question. A specific application scenario influenced me and I did overlook the more general case. Any item in the List[str] could be either text or an image. Xinference is responsible for returning the embedding vector for each object. As for the similarity between them, that’s up to the user. Does this align with Xinference's design?